### DISCUSSION TITLE
Unexpected results with PorousFlowSink

### DISCUSSION THREAD

**Discussion by nlosacco** [idaholab/moose#25613] [Q&A Modules: Porous Flow] [answered]
Hi all,

I am trying to simulate a desiccation test on an odometer (cylindrical) specimen by imposing an outflow boundary at the bases.
I simulate a quarter of the specimen diametral cross section in axisymmetric conditions, hence imposing the flow at the top only.
The specimen in initially fully saturated and with pore pressure = -10 kPa.
Given the specimen volume 2.46301E-05 m^3, the initial porosity 0.43, the imposed specific mass flow 1.47953E-07 kg/(m^2 . s) I expect the saturation ratio after 7 days to be 0.979, whereas MOOSE yield 0.9985 (and the pop remains -10 kPa).
The weirdest thing is that it nothing changes if I increase the flow by one order of magnitude!
Results start to change when I increase the flow by four orders of magnitude (which leads to convergence issues while the soil gets completely dry).
The sink BC seems to be applied correctly, as I checked using an AuxVariable and NodalSum postprocessing.
After trying for days I am quite lost. What am I missing?
Please find attached the input file.
[edo.i.txt](https://github.com/idaholab/moose/files/12748823/edo.i.txt)

**Comment by nlosacco** [comment]
As a matter of fact, I also postprocessed PorousFlowFluidMass, and the total mass is not changing at all

**Comment by GiudGiud** [comment]
@rpodgorney @1runer

**Comment by GiudGiud** [comment]
so the mass flow should be:
0.01 * 1 * 1.47953E-07 * 6 days in s ?
and the mass of fluid is 
0.01 * 0.028 * 1 * 1000 kg/m3 right ?

**Reply by nlosacco** [reply]
er... it's axisymmetric
xmax = 0.028 (radius)
ymax = 0.01 (height)
hence, if I'm not mistaken, the total mass flow should be:
pi * 0.028^2 * 1.47953E-07 * 604800 (7 days in seconds, I wrote 6 in the original post by mistake) = 2.2040E-04 kg
while the initial mass of fluid 
pi * 0.028^2 * 0.01 * 1000 * 0.43 (porosity) = 1.0591E-02 kg

**Comment by 1runer** [comment, marked as answer]
I'm not an expert on the BCs and all true/false parameters to play with. I guess @cpgr could help more.

What I noticed is that you don't solve your problem at all. Each timestep converges at 0 Nonlinear iterations. 
The residuals start with 1e-11 and your convergence criterion is 1e-7 (nl_abs_tol). I recommend changing the criterion or scale the variables accordingly (by setting automatic_scaling = true in the Executioner block.

**Reply by nlosacco** [reply]
That was the problem indeed. Now, setting nl_abs_tol = 1.E-12, the analysis is giving the expected results.
For larger problem I shall consider scaling the variables, so as not to use such a tight convergence criterion.
Thanks!
